### DISCUSSION TITLE
Integrate against third-party APIs using a database adapter

### DISCUSSION THREAD

**Discussion by junlarsen** [nextauthjs/next-auth#11429] [Help]
Hi, thanks for your work on Next Auth!

I'm working with the 5.0.0-beta releases of next-auth at the moment, and I'm trying to reproduce what is [done for the JWT strategy in the documentation](https://authjs.dev/guides/integrating-third-party-backends) for a setup that uses a database adapter instead. The adapter in question isn't important, but for sake of completeness I'll mention it's DynamoDB.

I use the AWS Cognito Provider where I'm particularly interested in using the Cognito-issued JWTs in a Next.js Route Handler (I'm looking to hit an external API using the AWS Cognito access token, which was granted to Next Auth on sign-in)

In the documentation using the JWT strategy, you use the jwt callback to attach tokens to the session. What would the equivalent look like for the database adapters?

Do I need to write glue code to grab the access tokens given a NextAuth session id from DynamoDB myself?

**Comment by k-taro56** [comment]
Yes, you need to write code to get access tokens from DynamoDB.
